pcsx2/Input/SDLInputSource.cpp

```diff
@@ -1519,6 +1519,28 @@ bool SDLInputSource::GetGenericBindingMapping(const std::string_view device, Inp
 	}
 }
 
+InputLayout SDLInputSource::GetControllerLayout(u32 index)
+{
+	// We are called from FSUI in the GS thread
+	std::lock_guard lock(m_controllers_key_mutex);
```

Comment:
Is this old?  I hope we don't call this from the GS thread, and it seems to only be referenced from InputManager::PreprocessEvent

Reply:
Must be, It took me a few attempts before I was happy with how the layout info made its way to the FSUI, so I may have added it during an earlier iteration.